### PR TITLE
fix: increase `Get-VCFSystemPrecheckTask` output verbosity

### LINKED ISSUE #255: Increase Get-VCFSystemPrecheckTask output verbosity to show failed subTasks
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.

### Description

Today Get-VCFSystemPrecheckTask shows truncated, unformatted output, reducing its usefulness.  One can only tell if the overall pre-checks process includes 0 or 1+ failed tasks.  

Two possible approaches, 

1. use `ConvertTo-Json` to uncompress the output.
2. Expand the output into an array and show only subTasks.stages.name where status == FAILED

### Naive Approach to option 1.

One could do something  change

`$response = Invoke-RestMethod -Method GET -Uri $uri -Headers $headers -ContentType 'application/json'`

to 

`$response = Invoke-RestMethod -Method GET -Uri $uri -Headers $headers -ContentType 'application/json' | ConvertTo-Json -Depth 2`

or 

something like:

`$body = $body | ConvertTo-Json -Depth 2`
`$response = Invoke-RestMethod -Method GET -Uri $uri -Headers $headers -ContentType 'application/json' -Body $body`

^ note the above isn't quite right as it seems to assume compressed output

For the invocation: `Get-VCFSystemPrecheckTask -id 8df688f2-b633-42d2-beca-0bcf43810089`        

This  `| ConvertTo-Json -Depth 2` addition changes the output of the commandlet from:

```powershell
id                  : 8df688f2-b633-42d2-beca-0bcf43810089
name                : PRECHECK
type                : PRECHECK
status              : FAILED
creationTimestamp   : 12/21/2023 9:18:40 PM
completionTimestamp : 12/21/2023 9:21:08 PM
subTasks            : {@{name=Upgrade - CLUSTER ; description=Upgrade - CLUSTER ; status=SUCCESSFUL; 
                      creationTimestamp=12/21/2023 9:18:40 PM; completionTimestamp=12/21/2023 
                      9:18:49 PM; stages=System.Object[]; resources=System.Object[]}, @{name=Upgrade - 
                      DEPLOYMENT_CONFIGURATION ; description=Upgrade - DEPLOYMENT_CONFIGURATION ; 
                      status=FAILED; creationTimestamp=12/21/2023 9:18:40 PM; 
                      completionTimestamp=12/21/2023 9:19:23 PM; stages=System.Object[]; 
                      resources=System.Object[]}, @{name=Upgrade - COMMON_SERVICES ; 
                      description=Upgrade - COMMON_SERVICES ; status=SUCCESSFUL; 
                      creationTimestamp=12/21/2023 9:19:23 PM; completionTimestamp=12/21/2023 
                      9:19:26 PM; stages=System.Object[]; resources=System.Object[]}, @{name=Upgrade - 
                      OPERATIONS_MANAGER ; description=Upgrade - OPERATIONS_MANAGER ; 
                      status=SUCCESSFUL; creationTimestamp=12/21/2023 9:18:40 PM; 
                      completionTimestamp=12/21/2023 9:18:41 PM; stages=System.Object[]; 
                      resources=System.Object[]}…}
isCancellable       : False
```
to 

```json      
{
  "id": "8df688f2-b633-42d2-beca-0bcf43810089",
  "name": "PRECHECK",
  "type": "PRECHECK",
  "status": "FAILED",
  "creationTimestamp": "2023-12-21T21:18:40.808Z",
  "completionTimestamp": "2023-12-21T21:21:08.175Z",
  "subTasks": [
    {
      "name": "Upgrade - CLUSTER ",
      "description": "Upgrade - CLUSTER ",
      "status": "SUCCESSFUL",
      "creationTimestamp": "2023-12-21T21:18:40.859Z",
      "completionTimestamp": "2023-12-21T21:18:49.074Z",
      "stages": " ",
      "resources": ""
    },
    {
      "name": "Upgrade - DEPLOYMENT_CONFIGURATION ",
      "description": "Upgrade - DEPLOYMENT_CONFIGURATION ",
      "status": "FAILED",
      "creationTimestamp": "2023-12-21T21:18:40.897Z",
      "completionTimestamp": "2023-12-21T21:19:23.975Z",
      "stages": "    ",
      "resources": ""
    },
    {
      "name": "Upgrade - COMMON_SERVICES ",
      "description": "Upgrade - COMMON_SERVICES ",
      "status": "SUCCESSFUL",
      "creationTimestamp": "2023-12-21T21:19:23.978Z",
      "completionTimestamp": "2023-12-21T21:19:26.898Z",
      "stages": "           ",
      "resources": ""
    },
    {
      "name": "Upgrade - OPERATIONS_MANAGER ",
      "description": "Upgrade - OPERATIONS_MANAGER ",
      "status": "SUCCESSFUL",
      "creationTimestamp": "2023-12-21T21:18:40.897Z",
      "completionTimestamp": "2023-12-21T21:18:41.334Z",
      "stages": "      ",
      "resources": ""
    },
    {
      "name": "Upgrade - SDDC_MANAGER_UI ",
      "description": "Upgrade - SDDC_MANAGER_UI ",
      "status": "SUCCESSFUL",
      "creationTimestamp": "2023-12-21T21:18:40.89Z",
      "completionTimestamp": "2023-12-21T21:18:41.332Z",
      "stages": "     ",
      "resources": ""
    },
    {
      "name": "Upgrade - DOMAIN_MANAGER ",
      "description": "Upgrade - DOMAIN_MANAGER ",
      "status": "SUCCESSFUL",
      "creationTimestamp": "2023-12-21T21:18:40.895Z",
      "completionTimestamp": "2023-12-21T21:18:41.349Z",
      "stages": "      ",
      "resources": ""
    },
    {
      "name": "Upgrade - LCM ",
      "description": "Upgrade - LCM ",
      "status": "WARNING",
      "creationTimestamp": "2023-12-21T21:19:26.905Z",
      "completionTimestamp": "2023-12-21T21:19:28.674Z",
      "stages": "         ",
      "resources": ""
    },
    {
      "name": "Upgrade - PSC ",
      "description": "Upgrade - PSC ",
      "status": "SUCCESSFUL",
      "creationTimestamp": "2023-12-21T21:18:40.906Z",
      "completionTimestamp": "2023-12-21T21:20:01.336Z",
      "stages": "       ",
      "resources": ""
    },
    {
      "name": "Upgrade - SDDC_SECURITY_CONFIGURATION ",
      "description": "Upgrade - SDDC_SECURITY_CONFIGURATION ",
      "status": "SUCCESSFUL",
      "creationTimestamp": "2023-12-21T21:18:41.79Z",
      "completionTimestamp": "2023-12-21T21:18:41.908Z",
      "stages": "  ",
      "resources": ""
    },
    {
      "name": "Upgrade - VCENTER ",
      "description": "Upgrade - VCENTER ",
      "status": "SUCCESSFUL",
      "creationTimestamp": "2023-12-21T21:18:40.927Z",
      "completionTimestamp": "2023-12-21T21:20:03.196Z",
      "stages": "         ",
      "resources": ""
    },
    {
      "name": "Upgrade - ESX ",
      "description": "Upgrade - ESX ",
      "status": "SUCCESSFUL",
      "creationTimestamp": "2023-12-21T21:19:28.693Z",
      "completionTimestamp": "2023-12-21T21:20:21.623Z",
      "stages": "               ",
      "resources": ""
    },
    {
      "name": "Upgrade - ESX ",
      "description": "Upgrade - ESX ",
      "status": "SUCCESSFUL",
      "creationTimestamp": "2023-12-21T21:19:28.682Z",
      "completionTimestamp": "2023-12-21T21:20:21.201Z",
      "stages": "               ",
      "resources": ""
    },
    {
      "name": "Upgrade - ESX ",
      "description": "Upgrade - ESX ",
      "status": "SUCCESSFUL",
      "creationTimestamp": "2023-12-21T21:19:28.686Z",
      "completionTimestamp": "2023-12-21T21:20:20.389Z",
      "stages": "               ",
      "resources": ""
    },
    {
      "name": "Upgrade - ESX ",
      "description": "Upgrade - ESX ",
      "status": "SUCCESSFUL",
      "creationTimestamp": "2023-12-21T21:19:28.686Z",
      "completionTimestamp": "2023-12-21T21:20:18.018Z",
      "stages": "               ",
      "resources": ""
    },
    {
      "name": "Upgrade - NSX_T ",
      "description": "Upgrade - NSX_T ",
      "status": "SUCCESSFUL",
      "creationTimestamp": "2023-12-21T21:18:44.645Z",
      "completionTimestamp": "2023-12-21T21:21:08.174Z",
      "stages": "                ",
      "resources": ""
    },
    {
      "name": "Upgrade - VSAN ",
      "description": "Upgrade - VSAN ",
      "status": "SUCCESSFUL",
      "creationTimestamp": "2023-12-21T21:18:48.324Z",
      "completionTimestamp": "2023-12-21T21:18:54.467Z",
      "stages": "                      ",
      "resources": ""
    }
  ],
  "isCancellable": false
}
```
### References

_No response_

### PR DESCRIPTION
**Pull request by nathanthaler**
### Summary

I've added "failureOnly" as an optional parameter to avoid breaking existing dependencies

Per @tenthirtyam 's request -- I've added a while loop to keep checking the pre-check task until it exits in-progress.  This applies to all invocations (not just ones with `-failureOnly`)

**Testing notes:**

```console
Remove-Module PowerVCF          
Import-Module ./PowerVCF.psm1 -Force
```
1. VCF 4.5.2 with one failed pre-check task (showing current behavior)

```console
> Get-VCFSystemPrecheckTask -id 1a25e644-3258-4d22-9cbf-ba58f92623d1  
```
```powershell
id                  : 1a25e644-3258-4d22-9cbf-ba58f92623d1
name                : PRECHECK
type                : PRECHECK
status              : FAILED
creationTimestamp   : 1/2/2024 12:34:48 PM
completionTimestamp : 1/2/2024 12:37:18 PM
subTasks            : {@{name=Upgrade - CLUSTER ; description=Upgrade - CLUSTER ; status=SUCCESSFUL; creationTimestamp=1/2/2024 12:34:48 PM; completionTimestamp=1/2/2024 12:34:50 PM; stages=System.Object[]; 
                      resources=System.Object[]}, @{name=Upgrade - DEPLOYMENT_CONFIGURATION ; description=Upgrade - DEPLOYMENT_CONFIGURATION ; status=FAILED; creationTimestamp=1/2/2024 12:34:49 PM; 
                      completionTimestamp=1/2/2024 12:35:23 PM; stages=System.Object[]; resources=System.Object[]}, @{name=Upgrade - COMMON_SERVICES ; description=Upgrade - COMMON_SERVICES ; status=SUCCESSFUL; 
                      creationTimestamp=1/2/2024 12:35:23 PM; completionTimestamp=1/2/2024 12:35:26 PM; stages=System.Object[]; resources=System.Object[]}, @{name=Upgrade - OPERATIONS_MANAGER ; description=Upgrade - 
                      OPERATIONS_MANAGER ; status=SUCCESSFUL; creationTimestamp=1/2/2024 12:34:49 PM; completionTimestamp=1/2/2024 12:34:49 PM; stages=System.Object[]; resources=System.Object[]}…}
isCancellable       : False
```

2. VCF 4.5.2 with one failed pre-check task (exercising new option -- failed only output)
```console
> Get-VCFSystemPrecheckTask -id 1a25e644-3258-4d22-9cbf-ba58f92623d1 -failureOnly
```
```powershell
VCF System Precheck complete. Overall Status: FAILED

name                : SDDC_MANAGER_FILE_PERMISSIONS_CHECK
description         : Perform Stage - Checks permissions of files and directories used by SDDC Manager services.
status              : FAILED
creationTimestamp   : 1/2/2024 12:34:49 PM
completionTimestamp : 1/2/2024 12:34:51 PM
errors              : {}
```

3. VCF 5.0.0 with pre-check running (showing current behavior)
```console
> Get-VCFSystemPrecheckTask -id 5cb59bbc-4e4d-44db-9d62-d37f7bdaf684
```             
```powershell
id                : 5cb59bbc-4e4d-44db-9d62-d37f7bdaf684
name              : PRECHECK
type              : PRECHECK
status            : IN_PROGRESS
creationTimestamp : 1/2/2024 1:12:02 PM
subTasks          : {@{name=Upgrade - CLUSTER ; description=Upgrade - CLUSTER ; status=SUCCESSFUL; creationTimestamp=1/2/2024 1:12:02 PM; completionTimestamp=1/2/2024 1:12:04 PM; stages=System.Object[]; resources=System.Object[]}, @{name=Upgrade - 
                    DEPLOYMENT_CONFIGURATION ; description=Upgrade - DEPLOYMENT_CONFIGURATION ; status=FAILED; creationTimestamp=1/2/2024 1:12:02 PM; completionTimestamp=1/2/2024 1:12:45 PM; stages=System.Object[]; resources=System.Object[]}, 
                    @{name=Upgrade - COMMON_SERVICES ; description=Upgrade - COMMON_SERVICES ; status=SUCCESSFUL; creationTimestamp=1/2/2024 1:12:45 PM; completionTimestamp=1/2/2024 1:12:48 PM; stages=System.Object[]; resources=System.Object[]}, 
                    @{name=Upgrade - OPERATIONS_MANAGER ; description=Upgrade - OPERATIONS_MANAGER ; status=SUCCESSFUL; creationTimestamp=1/2/2024 1:12:02 PM; completionTimestamp=1/2/2024 1:12:02 PM; stages=System.Object[]; 
                    resources=System.Object[]}…}
isCancellable     : False
```
4. VCF 5.0.0 with pre-check complete (showing current behavior)
```console
> Get-VCFSystemPrecheckTask -id 5cb59bbc-4e4d-44db-9d62-d37f7bdaf684            
``` 
```powershell
id                  : 5cb59bbc-4e4d-44db-9d62-d37f7bdaf684
name                : PRECHECK
type                : PRECHECK
status              : FAILED
creationTimestamp   : 1/2/2024 1:12:02 PM
completionTimestamp : 1/2/2024 1:14:24 PM
subTasks            : {@{name=Upgrade - CLUSTER ; description=Upgrade - CLUSTER ; status=SUCCESSFUL; creationTimestamp=1/2/2024 1:12:02 PM; completionTimestamp=1/2/2024 1:12:04 PM; stages=System.Object[]; resources=System.Object[]}, @{name=Upgrade - 
                      DEPLOYMENT_CONFIGURATION ; description=Upgrade - DEPLOYMENT_CONFIGURATION ; status=FAILED; creationTimestamp=1/2/2024 1:12:02 PM; completionTimestamp=1/2/2024 1:12:45 PM; stages=System.Object[]; resources=System.Object[]}, 
                      @{name=Upgrade - COMMON_SERVICES ; description=Upgrade - COMMON_SERVICES ; status=SUCCESSFUL; creationTimestamp=1/2/2024 1:12:45 PM; completionTimestamp=1/2/2024 1:12:48 PM; stages=System.Object[]; resources=System.Object[]}, 
                      @{name=Upgrade - OPERATIONS_MANAGER ; description=Upgrade - OPERATIONS_MANAGER ; status=SUCCESSFUL; creationTimestamp=1/2/2024 1:12:02 PM; completionTimestamp=1/2/2024 1:12:02 PM; stages=System.Object[]; 
                      resources=System.Object[]}…}
isCancellable       : False
```
5. VCF 5.0.0 with pre-check still running and two failures thus far (exercising new option -- failed only output)
```console
> Get-VCFSystemPrecheckTask -id 5cb59bbc-4e4d-44db-9d62-d37f7bdaf684 -failureOnly          
```                         
```powershell
VCF System Precheck still in progress. Please re-run to see the complete results

name                : Network - vMotion: Basic (unicast) connectivity check
description         : Perform Stage - Network - vMotion: Basic (unicast) connectivity check
status              : FAILED
creationTimestamp   : 1/2/2024 1:12:04 PM
completionTimestamp : 1/2/2024 1:12:04 PM
errors              : {}

name                : Network - vMotion: MTU check (ping with large packet size)
description         : Perform Stage - Network - vMotion: MTU check (ping with large packet size)
status              : FAILED
creationTimestamp   : 1/2/2024 1:12:04 PM
completionTimestamp : 1/2/2024 1:12:04 PM
errors              : {}
```

6. VCF 5.0.0 with pre-check complete and multiple failures (exercising new option -- failed only output)
```console
> Get-VCFSystemPrecheckTask -id 5cb59bbc-4e4d-44db-9d62-d37f7bdaf684 -failureOnly                                   
```
```powershell
VCF System Precheck complete. Overall Status: FAILED

name                : SDDC_MANAGER_FILE_PERMISSIONS_CHECK
description         : Perform Stage - Checks permissions of files and directories used by SDDC Manager services.
status              : FAILED
creationTimestamp   : 1/2/2024 1:12:44 PM
completionTimestamp : 1/2/2024 1:12:45 PM
errors              : {}

name                : LCM_MANIFEST_OUT_OF_DATE_CHECK
description         : Perform Stage - Checks if LCM manifest is out of date.
status              : FAILED
creationTimestamp   : 1/2/2024 1:12:49 PM
completionTimestamp : 1/2/2024 1:12:49 PM
errors              : {}

name                : Network - vMotion: Basic (unicast) connectivity check
description         : Perform Stage - Network - vMotion: Basic (unicast) connectivity check
status              : FAILED
creationTimestamp   : 1/2/2024 1:12:04 PM
completionTimestamp : 1/2/2024 1:12:04 PM
errors              : {}

name                : Network - vMotion: MTU check (ping with large packet size)
description         : Perform Stage - Network - vMotion: MTU check (ping with large packet size)
status              : FAILED
creationTimestamp   : 1/2/2024 1:12:04 PM
completionTimestamp : 1/2/2024 1:12:04 PM
errors              : {}
```

6. VCF 5.0.0 with pre-check complete with no failures (showing current behavior)
```console
> Get-VCFSystemPrecheckTask -id 5cb59bbc-4e4d-44db-9d62-d37f7bdaf684
```
```powershell
id                  : 5cb59bbc-4e4d-44db-9d62-d37f7bdaf684
name                : PRECHECK
type                : PRECHECK
status              : SUCCESSFUL
creationTimestamp   : 1/2/2024 1:12:02 PM
completionTimestamp : 1/2/2024 1:14:24 PM
subTasks            : {@{name=Upgrade - CLUSTER ; description=Upgrade - CLUSTER ; status=SUCCESSFUL; creationTimestamp=1/2/2024 1:12:02 PM; completionTimestamp=1/2/2024 1:12:04 PM; stages=System.Object[]; resources=System.Object[]}, @{name=Upgrade - 
                      DEPLOYMENT_CONFIGURATION ; description=Upgrade - DEPLOYMENT_CONFIGURATION ; status=SUCCESSFUL; creationTimestamp=1/2/2024 1:12:02 PM; completionTimestamp=1/2/2024 1:12:45 PM; stages=System.Object[]; resources=System.Object[]}, 
                      @{name=Upgrade - COMMON_SERVICES ; description=Upgrade - COMMON_SERVICES ; status=SUCCESSFUL; creationTimestamp=1/2/2024 1:12:45 PM; completionTimestamp=1/2/2024 1:12:48 PM; stages=System.Object[]; resources=System.Object[]}, 
                      @{name=Upgrade - OPERATIONS_MANAGER ; description=Upgrade - OPERATIONS_MANAGER ; status=SUCCESSFUL; creationTimestamp=1/2/2024 1:12:02 PM; completionTimestamp=1/2/2024 1:12:02 PM; stages=System.Object[]; 
                      resources=System.Object[]}…}
isCancellable       : False
```
7. VCF 5.0.0 with pre-check complete with no failures (exercising new option -- failed only output)
```console
> Get-VCFSystemPrecheckTask -id 5cb59bbc-4e4d-44db-9d62-d37f7bdaf684 -failureOnly
```
```powershell
id                  : 5cb59bbc-4e4d-44db-9d62-d37f7bdaf684
name                : PRECHECK
type                : PRECHECK
status              : SUCCESSFUL
creationTimestamp   : 1/2/2024 1:12:02 PM
completionTimestamp : 1/2/2024 1:14:24 PM
subTasks            : {@{name=Upgrade - CLUSTER ; description=Upgrade - CLUSTER ; status=SUCCESSFUL; creationTimestamp=1/2/2024 1:12:02 PM; completionTimestamp=1/2/2024 1:12:04 PM; stages=System.Object[]; resources=System.Object[]}, @{name=Upgrade - 
                      DEPLOYMENT_CONFIGURATION ; description=Upgrade - DEPLOYMENT_CONFIGURATION ; status=SUCCESSFUL; creationTimestamp=1/2/2024 1:12:02 PM; completionTimestamp=1/2/2024 1:12:45 PM; stages=System.Object[]; resources=System.Object[]}, 
                      @{name=Upgrade - COMMON_SERVICES ; description=Upgrade - COMMON_SERVICES ; status=SUCCESSFUL; creationTimestamp=1/2/2024 1:12:45 PM; completionTimestamp=1/2/2024 1:12:48 PM; stages=System.Object[]; resources=System.Object[]}, 
                      @{name=Upgrade - OPERATIONS_MANAGER ; description=Upgrade - OPERATIONS_MANAGER ; status=SUCCESSFUL; creationTimestamp=1/2/2024 1:12:02 PM; completionTimestamp=1/2/2024 1:12:02 PM; stages=System.Object[]; 
                      resources=System.Object[]}…}
isCancellable       : False
```

### Type

- [ ] Bugfix
- [x] Enhancement or Feature
- [ ] Code Style or Formatting
- [ ] Documentation
- [ ] Refactoring
- [ ] Chore
- [ ] Other
        Please describe:

### Breaking Changes?

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### Test and Documentation

- [x] Tests have been completed.
- [x] Documentation has been added or updated.

### Issue References

Closes #255

### CODE REVIEW

File: PowerVCF.psm1 [hunk 1]
```diff
@@ -4258,10 +4258,10 @@ Export-ModuleMember -Function Start-VCFSystemPrecheck
 Function Get-VCFSystemPrecheckTask {
     <#
         .SYNOPSIS
-        Retrieves the status of a system level precheck task.
+        Retrieves the status of a system level precheck task and returns the details on failed subtasks
```
Comment (tenthirtyam) [inline PowerVCF.psm1]:
```suggestion
        Retrieves the status of a system level precheck task and returns the details on failed subtasks.
```

Comment (tenthirtyam) [reply PowerVCF.psm1]: I think this could be left as-is since the new parameter is optional.

File: PowerVCF.psm1 [hunk 2]
```diff
@@ -4270,18 +4270,37 @@ Function Get-VCFSystemPrecheckTask {
...
     Try {
         createHeader # Set the Accept and Authorization headers.
         checkVCFToken # Validate the access token and refresh, if necessary.
-        $uri = "https://$sddcManager/v1/system/prechecks/tasks/$id"
-        $response = Invoke-RestMethod -Method GET -Uri $uri -Headers $headers -ContentType 'application/json'
-        $response
+        
+        Do {
+            # Keep checking until status is not IN_PROGRESS
+            $uri = "https://$sddcManager/v1/system/prechecks/tasks/$id"
```
Comment (tenthirtyam) [inline PowerVCF.psm1]: I would leave the uri in it's previous location.

File: docs/documentation/functions/system-prechecks/Get-VCFSystemPrecheckTask.md [hunk 3]
```diff
@@ -24,6 +24,13 @@ Get-VCFSystemPrecheckTask -id 4d661acc-2be6-491d-9256-ba3c78020e5d
 
 This example shows how to retrieve the status of a system level precheck task by unique ID.
 
+### Example 2 
+```powershell
+Get-VCFSystemPrecheckTask -id 4d661acc-2be6-491d-9256-ba3c78020e5d -failedonly
+```
+
+This example shows how to retrieve only failed subtasks from the system level precheck task by unique ID.
+
```
Comment (tenthirtyam) [inline docs/documentation/functions/system-prechecks/Get-VCFSystemPrecheckTask.md]: This should also be the second `.EXAMPLE` as seen in the cmdlet help in the `.psm1`.

File: PowerVCF.psm1 [hunk 4]
```diff
@@ -4270,18 +4270,37 @@ Function Get-VCFSystemPrecheckTask {
```
Comment (tenthirtyam) [inline PowerVCF.psm1:4270]:
```suggestion

        Get-VCFSystemPrecheckTask -id 4d661acc-2be6-491d-9256-ba3c78020e5d -failedonly
        This example shows how to retrieve only failed subtasks from the system level precheck task by unique ID.

```

File: PowerVCF.psm1 [hunk 5]
```diff
@@ -4270,18 +4270,37 @@ Function Get-VCFSystemPrecheckTask {
 
         .PARAMETER id
         Specifies the unique ID of the system level precheck task.
+
+        .PARAMETER failedonly
+        Optional: only show failed subtasks
+
```
Comment (tenthirtyam) [inline PowerVCF.psm1]:
```suggestion
        .PARAMETER failedonly
        Specifies to return only the failed subtasks.
```

File: docs/documentation/functions/system-prechecks/Get-VCFSystemPrecheckTask.md [hunk 6]
```diff
@@ -41,7 +48,19 @@ Default value: None
 Accept pipeline input: False
 Accept wildcard characters: False
 ```
+### -failedonly
```
Comment (tenthirtyam) [inline docs/documentation/functions/system-prechecks/Get-VCFSystemPrecheckTask.md]:
```suggestion

### -failedonly

Specifies to return only the failed subtasks.
```

File: docs/documentation/functions/system-prechecks/Get-VCFSystemPrecheckTask.md [hunk 7]
```diff
@@ -41,7 +48,19 @@ Default value: None
...
+Type: Switch
+Parameter Sets: (All)
+Aliases:
 
+Required: False
+Position: 2
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
```
Comment (tenthirtyam) [inline docs/documentation/functions/system-prechecks/Get-VCFSystemPrecheckTask.md:66]:
```suggestion
```

```

File: PowerVCF.psm1 [hunk 8]
```diff
@@ -4270,18 +4270,37 @@ Function Get-VCFSystemPrecheckTask {
...
         checkVCFToken # Validate the access token and refresh, if necessary.
         $uri = "https://$sddcManager/v1/system/prechecks/tasks/$id"
-        $response = Invoke-RestMethod -Method GET -Uri $uri -Headers $headers -ContentType 'application/json'
-        $response
+    
+        Do {
+            # Keep checking until status is not IN_PROGRESS
+            $response = Invoke-RestMethod -Method GET -Uri $uri -Headers $headers -ContentType 'application/json'
+            } While ($response.status -eq "IN_PROGRESS") 
+                Start-Sleep -Seconds 2
```
Comment (tenthirtyam) [inline PowerVCF.psm1]:
```suggestion
        } While ($response.status -eq "IN_PROGRESS")
```

File: PowerVCF.psm1 [hunk 9]
```diff
@@ -4270,18 +4270,37 @@ Function Get-VCFSystemPrecheckTask {
...
     Try {
         createHeader # Set the Accept and Authorization headers.
         checkVCFToken # Validate the access token and refresh, if necessary.
         $uri = "https://$sddcManager/v1/system/prechecks/tasks/$id"
-        $response = Invoke-RestMethod -Method GET -Uri $uri -Headers $headers -ContentType 'application/json'
-        $response
+    
+        Do {
+            # Keep checking until status is not IN_PROGRESS
+            $response = Invoke-RestMethod -Method GET -Uri $uri -Headers $headers -ContentType 'application/json'
```
Comment (tenthirtyam) [inline PowerVCF.psm1:4295]:
```suggestion
            $response = Invoke-RestMethod -Method GET -Uri $uri -Headers $headers -ContentType 'application/json'
            Start-Sleep -Seconds 2
```

Added `Start-Sleep -Seconds 2`` command inside the loop to pause for 2 seconds between each request.

File: PowerVCF.psm1 [hunk 10]
```diff
@@ -4270,18 +4270,37 @@ Function Get-VCFSystemPrecheckTask {
...
+            } While ($response.status -eq "IN_PROGRESS") 
+                Start-Sleep -Seconds 2
+
+        if ($PsBoundParameters.ContainsKey("failedonly")) {
+            Write-Output "VCF System Precheck complete. Overall Status: $($response.status)"
```
Comment (tenthirtyam) [inline PowerVCF.psm1]:
```suggestion
            Write-Output "System Precheck complete. Overall Status: $($response.status)"
```

**Review by tenthirtyam — COMMENTED** [review]
I’m thinking you could do something like this:
 
 ```powershell
 Do {
    # Keep checking until status is not IN_PROGRESS
    $response = Invoke-RestMethod -Method GET -Uri $uri -Headers $headers -ContentType 'application/json'
    Start-Sleep -Seconds 2
} While ($response.status -eq "IN_PROGRESS")

if ($response.status -eq "FAILED") {
    if ($PsBoundParameters.ContainsKey("failedonly")) {
        $failed_task = $response.subTasks | Where-Object { $_.status -eq "FAILED" }
        $failed_subtask = $failed_task.stages | Where-Object { $_.status -eq "FAILED" }
        $failed_subtask
    } else {
        $response
    }
} else  {
    $response
}
```

**Comment by tenthirtyam** [comment]
Or...

```powershell
Do {
    # Keep checking until status is not IN_PROGRESS
    $response = Invoke-RestMethod -Method GET -Uri $uri -Headers $headers -ContentType 'application/json'
    Start-Sleep -Seconds 2
} While ($response.status -eq "IN_PROGRESS")

if ($response.status -eq "FAILED" -and $PsBoundParameters.ContainsKey("failedonly")) {
    $failed_task = $response.subTasks | Where-Object { $_.status -eq "FAILED" }
    $failed_subtask = $failed_task.stages | Where-Object { $_.status -eq "FAILED" }
    $failed_subtask
} else {
    $response
}
```

File: docs/documentation/functions/system-prechecks/Get-VCFSystemPrecheckTask.md [hunk 11]
```diff
@@ -24,6 +24,13 @@ Get-VCFSystemPrecheckTask -id 4d661acc-2be6-491d-9256-ba3c78020e5d
 
 This example shows how to retrieve the status of a system level precheck task by unique ID.
 
+### Example 2 
+```powershell
+Get-VCFSystemPrecheckTask -id 4d661acc-2be6-491d-9256-ba3c78020e5d -failedonly
+```
```
Comment (tenthirtyam) [inline docs/documentation/functions/system-prechecks/Get-VCFSystemPrecheckTask.md]:
```suggestion
### Example 2

```powershell
Get-VCFSystemPrecheckTask -id 4d661acc-2be6-491d-9256-ba3c78020e5d -failedonly
```
```

Comment (nathanthaler) [reply docs/documentation/functions/system-prechecks/Get-VCFSystemPrecheckTask.md]: I believe this is fixed in the latest commit

**Review by tenthirtyam — APPROVED** [review]
LGTM! 🚀

**Pull request merged**